File: gettls/week12/신규 아이디 추천.cpp
#include <string>
#include <vector>

using namespace std;

string solution(string new_id) {
    string answer = "";

    for (int i = 0; i < new_id.size(); i++) {
        if ((new_id[i] >= 'a' && new_id[i] <= 'z')
            || (new_id[i] >= '0' && new_id[i] <= '9')
            || new_id[i] == '-' || new_id[i] == '_' || new_id[i]=='.') continue;
        else if (new_id[i] >= 'A' && new_id[i] <= 'Z') new_id[i] = new_id[i] - 'A' + 'a';
        else {
            new_id.replace(i, 1, "");
            i--;
        }
    }

    for (int i = 0; i < new_id.size(); i++) {
        if (new_id[i] == '.') {
            int j = 0;
            while (new_id[i+j] == '.') j++;
            if (j != 1) {
                new_id.replace(i, j, ".");
                i--;
            }
        }
    }
    if (new_id[0] == '.') new_id.replace(0, 1, "");
    if (new_id[new_id.size() - 1] == '.') new_id.replace(new_id.size() - 1, 1, "");

    answer = new_id;
    if (answer.size() >= 16) {
        answer = answer.substr(0, 15);
        while (answer[answer.size() - 1] == '.') answer.replace(answer.size() - 1, 1, "");
        while (answer[0] == '.') answer.replace(0, 1, "");
    }
    if (answer.size() <= 2) {
        if (answer.size() == 0) answer = "aaa";
        else while (answer.size() < 3) answer += answer[answer.size() - 1];
    }


    return answer;
}

File: wonseok970/2021 KAKAO BLIND RECRUITMENT/신규 아이디 추천.py
def solution(new_id):
    answer = ''
    # 1.
    new_id = new_id.lower()

    # 2.
    for i in new_id:
        if i.isalnum() or i in '-_.':
            answer += i

    # 3.
    while '..' in answer:
        answer = answer.replace('..', '.')

    # 4.
    if answer[0] == '.' and len(answer) > 1:
        answer = answer[1:]
    else:
        pass

    if answer[-1] == '.':
        answer = answer[:-1]

    # 5.
    if len(answer) == 0:
        answer = 'a'

    # 6.
    if len(answer) >= 16:
        answer = answer[:15]
        if answer[0] == '.' and len(answer) > 1:
            answer = answer[1:]
        else:
            pass

        if answer[-1] == '.':
            answer = answer[:-1]

    # 7.
    if len(answer) <= 3:
        answer += answer[-1] * (3 - len(answer))

    return answer